.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,104 @@
+name: Trigger image scanning on pull request and changes to pr
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+    branches: [main]
+
+jobs:
+  deploy:
+    name: Trigger image scanning
+    runs-on: ubuntu-latest
+    env:
+      PEAK_USER_ID: 8877
+      BRANCH: ${{ github.head_ref }}
+      SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+      GH_PAGES_BASE_URL: https://peak-ai.github.io/platform-global-images
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get changed directories
+        id: changed-directories
+        uses: tj-actions/changed-files@v23
+        with:
+          files_ignore: |
+            *.md
+            knowledge-base
+            .github/workflows
+            .gitignore
+            license
+          dir_names: true
+
+      - name: List all changed directories/images
+        run: |
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            echo "$directory was changed"
+          done
+
+      - name: Run docker build on changed images
+        if: steps.changed-directories.outputs.any_changed == 'true'
+        shell: bash --noprofile --norc {0}
+        run: |
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            tag=$(echo $directory | tr / -)
+            echo "Building $directory"
+            docker build $directory --build-arg PEAK_USER_ID=$PEAK_USER_ID -t $tag
+          done
+
+      - name: Run snyk scan on changed images
+        if: steps.changed-directories.outputs.any_changed == 'true'
+        shell: bash --noprofile --norc {0}
+        id: snyk-scans
+        run: |
+          npx snyk auth $SNYK_TOKEN
+          mkdir -p /tmp/$BRANCH
+          scan_summary="<table><tr><th>Image name</th><th>Critical</th><th>High</th><th>Medium</th><th>Low</th><th>Report</th></tr>"
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            tag=$(echo $directory | tr / -)
+            docker image inspect $tag > /dev/null 2>&1
+
+            if [[ $? -ne 0 ]]; then
+              scan_summary="$scan_summary <tr><td>$tag</td><td>-</td><td>-</td><td>-</td><td>-</td><td>Image Build Failed</td></tr>"
+              continue
+            fi
+
+            npx snyk container test $tag --json >> $tag.json
+            tag_json=`cat $tag.json`
+            npx snyk-to-html -i $tag.json -o /tmp/$BRANCH/$tag.html -d
+            all_vulnerabilities=`jq -r '.vulnerabilities[].severity' $tag.json`
+            critical=`echo $all_vulnerabilities | grep -wo "critical" | wc -l | xargs`
+            high=`echo $all_vulnerabilities | grep -wo "high" | wc -l | xargs`
+            medium=`echo $all_vulnerabilities | grep -wo "medium" | wc -l | xargs`
+            low=`echo $all_vulnerabilities | grep -wo "low" | wc -l | xargs`
+            report_url=$(echo "<a href='$GH_PAGES_BASE_URL/$BRANCH/$tag.html' target='_blank'>Click here</a>")
```

Comment:
this doesn't seem to be opening the results in a new tab.

Reply:
Sadly GitHub doesn't seem to support opening URL in a new tab 😢